### ISSUE TITLE
FR: Flex SDR Radio support

### ISSUE THREAD

**Issue by n61ab** [tailscale/tailscale#4772]
### What are you trying to do?

I have a Flex SDR Radio on my network that uses UDP to transport a discovery packet and to transport the radio information.  I want to access the Flex radio using Tailscale sub net route.

Problem, it appears that Tailscale does not bridge the level two packets allowing the UDP packets to move across the WAN.

### How should we solve this?

Add an option for Level Two bridging.

### What is the impact of not solving this?

I cannot access the UDP packets via Tailscale sub net routes.

**Comment by DentonGentry** [comment]
"discovery packet" means it is sent to a multicast destination? Is it mDNS, or something custom?

mDNS support is tracked in https://github.com/tailscale/tailscale/issues/1013. We're likely to rely on relays to forward packets, rather than begin forwarding all multicasts across the WAN.

**Comment by n61ab** [comment]
Hi Denton

Here is a link to a Wiki describing the UDP protocols for my Flex SDR Radio.

https://github.com/flexradio/smartsdr-api-docs/wiki/SmartSDR-Ethernet-API

I am referring to items two and three.

Alan

**Comment by DentonGentry** [comment]
From those links I can't figure out what destination IP address the UDP packets will be sent to. If it is a multicast address, it won't be forwarded over the tailnet automatically and would need a relay or proxy. https://github.com/tailscale/tailscale/issues/1013 describes similar plans for mDNS.

If the SDR is sending UDP frames unicast to a specific destination, then my guess would be it is running into an MTU issue. The MTU size through the Wireguard tunnel is 1280 bytes. If the SDR is sending 1460 byte frames for Ethernet, those frames would need to be fragmented to make it over the tunnel. If the dont-fragment bit is set (which is quite possible as fragmentation on the whole isn't a great thing) then the frames will just be dropped.

If the problem is with the MTU, hopefully there is a setting to reduce the maximum size of packets sent.

**Comment by n61ab** [comment]
Denton

Both.

The "Discovery" UDP packet, port 4992, item three in the WiKi, is multicast.

The "Steaming Data" UDP packets, port 4991, item two in the Wiki, are sent to a specific IP.

The "Discovery Packet" sends out the IP address and name of the radio to the network.  The client application uses this information to bind to the radio (server) via the provided IP.

MTU of the "Streaming Data" is adjustable via a TCP connection from the client.

So...for now the first problem is getting the UDP port 4992 multicast packets across the WAN.

Next issue is allowing the Streaming UDP packets to a specific IP across the WAN

Alan

**Comment by DentonGentry** [comment]
The unicast UDP packets should work over the tailnet now. UDP works. If the packets are larger than the 1280 byte MTU size, they may be hitting a problem with the packet size.

**Comment by n61ab** [comment]
Denton, thanks.  So....the discovery shout work....good.

So far, I have not been able to get the discovery packets to go over Tailscale sub net route.  Let me spend some time testing further and I will get back to you.

**Comment by DentonGentry** [comment]
> The "Discovery" UDP packet, port 4992, item three in the WiKi, is multicast.
...
> So....the discovery shout work....good.

Multicast is not forwarded over the tailnet. Multicast packets would need a relay of some sort. I do not expect the multicast Discovery packets to work.

Unicast UDP packets will work. I expect the Streaming Data UDP packets to work, if you can get it to the point of sending data without having the discovery packets.

**Comment by n61ab** [comment]
Denton - Got it working!

Using Node-Red, I built a UDP bridges on the server and the client.  

On the Server Side, I bridged the discovery packet, port 4992 to port 7000 to the Client's Tailscale IP address.

On the Client Side, I bridged the UDP heard on port 7000, back to port 4992, Multicast, which the Flex SDR application was listening.

Now, my Flex Smart SDR client application sees the discovery packets, from the Server, via Tailscale WAN....and the application has the data it needs to connect.

After connecting, with MTU 1490, the streaming UDP, port 4991 packets did not get through, as you told me.  I reduced the MTU to 1200, and now....everything works.

Alan
